src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/SerializationWrapperForStream.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Text.Json.Serialization.Metadata;
+using System.Threading.Tasks;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    /// <summary>
+    /// Base class for wrapping serialization calls which allows tests to run under different configurations.
+    /// This emulates the JsonSerializer stream-based APIs directly which is a different approach than the
+    /// SerializationWrapper derived classes which emulate the JsonSerializer string-based APIs.
+    /// </summary>
+    public abstract class SerializationWrapperForStream
```

Comment:
nit: name this so that it's clearer that it supports both serialization and deserialization? I'd suggest renaming `SerializationWrapper` -> `JsonSerializerWrapper` as well

```suggestion
    public abstract class JsonSerializerWrapperForStream
```